src/Adapter/MerchandiseReturn/AbstractMerchandiseReturnHandler.php

```diff
@@ -0,0 +1,138 @@
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\MerchandiseReturn;
+
+use OrderReturn;
+use PrestaShop\PrestaShop\Adapter\Domain\AbstractObjectModelHandler;
+use PrestaShop\PrestaShop\Core\Domain\MerchandiseReturn\Exception\DeleteMerchandiseReturnDetailException;
+use PrestaShop\PrestaShop\Core\Domain\MerchandiseReturn\Exception\MerchandiseReturnException;
+use PrestaShop\PrestaShop\Core\Domain\MerchandiseReturn\Exception\MerchandiseReturnNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\MerchandiseReturn\Exception\MissingMerchandiseReturnRequiredFieldsException;
+use PrestaShop\PrestaShop\Core\Domain\MerchandiseReturn\ValueObject\CustomizationId;
+use PrestaShop\PrestaShop\Core\Domain\MerchandiseReturn\ValueObject\MerchandiseReturnDetailId;
+use PrestaShop\PrestaShop\Core\Domain\MerchandiseReturn\ValueObject\MerchandiseReturnId;
+use PrestaShopException;
+
+/**
+ * Provides reusable methods for merchandise return command/query handlers
+ */
+abstract class AbstractMerchandiseReturnHandler extends AbstractObjectModelHandler
+{
+    /**
+     * @param MerchandiseReturnId $merchandiseReturnId
+     * @param OrderReturn $orderReturn
+     *
+     * @throws MerchandiseReturnNotFoundException
+     */
+    protected function assertOrderReturnWasFound(MerchandiseReturnId $merchandiseReturnId, OrderReturn $orderReturn): void
+    {
+        if ($orderReturn->id !== $merchandiseReturnId->getValue()) {
+            throw new MerchandiseReturnNotFoundException($merchandiseReturnId, sprintf('Order return with id "%s" was not found.', $merchandiseReturnId->getValue()));
+        }
+    }
+
+    /**
+     * Gets legacy OrderReturn
+     *
+     * @param MerchandiseReturnId $merchandiseReturnId
+     *
+     * @return OrderReturn
+     *
+     * @throws MerchandiseReturnException
+     */
+    protected function getOrderReturn(MerchandiseReturnId $merchandiseReturnId): OrderReturn
+    {
+        try {
+            $orderReturn = new OrderReturn($merchandiseReturnId->getValue());
+        } catch (PrestaShopException $e) {
+            throw new MerchandiseReturnException('Failed to create new order return', 0, $e);
+        }
+
+        if ($orderReturn->id !== $merchandiseReturnId->getValue()) {
+            throw new MerchandiseReturnNotFoundException($merchandiseReturnId, sprintf('Merchandise return with id "%s" was not found.', $merchandiseReturnId->getValue()));
+        }
+
+        return $orderReturn;
+    }
+
+    protected function deleteMerchandiseReturnDetail(
+        MerchandiseReturnId $merchandiseReturnId,
+        MerchandiseReturnDetailId $merchandiseReturnDetailId,
+        ?CustomizationId $customizationId
+    ) {
+        $orderReturn = new OrderReturn($merchandiseReturnId->getValue());
+        $this->assertOrderReturnWasFound($merchandiseReturnId, $orderReturn);
+
+        /**
+         * @todo I am not sure if this is the right solution, but one alternative was do do an if
```

Comment:
This is fine to me as long we clearly explain what we are doing.
So your explanation here is fine and must be kept 😛 

2 solutions:
- keep it as phpDoc here but it's a bit long, not sure people will read it 😅 
- use code structure to make it very clear, example below ⬇️ 

```
if ($customizationId !== null) {
    $this->deleteReturnDetailWithCustomisation();
} else {
    $this->deleteReturnDetail();
}
```

Don't be afraid of creating private functions with very clear names. You can duplicate 2 to 5 lines - it's OK if the function name carries a critical information. Code readability is more important in a project as important as PrestaShop than saving some lines.

Reply:
Other possibility: use SWITCH with very clear usecases. It's another very clear way to explain different usecases.

Reply:
I think solution with if makes the most sense. I don't really see the benefit of using SWITCH here over that.